### PR TITLE
Hyrax Orcid Gem Integration

### PR DESCRIPTION
**Pull request by prdanelli**
Integrate the Hyrax Orcid Gem into Hyku Addons

### CODE REVIEW

File: lib/hyku_addons/engine.rb [hunk 1]
```diff
@@ -513,6 +519,9 @@ def self.dynamically_include_mixins
       # Insert at the end of the actor chain
       Hyrax::CurationConcern.actor_factory.use HykuAddons::Actors::TaskMaster::WorkActor
 
+      # Remove the Hyrax Orcid JSON Actor as we have our own
+      Hyrax::CurationConcern.actor_factory.middlewares.delete(::Hyrax::Actors::Orcid::JSONFieldsActor)
```
Comment (mankind) [inline lib/hyku_addons/engine.rb:523]: 1. I can't see **Hyrax::Actors::Orcid::JSONFieldsActor** in hyrax or hyku_addons. Instead i saw one with a different name here:  **HykuAddons::Actors::Orcid::JSONFieldsActor**
https://github.com/ubiquitypress/hyku_addons/blob/main/app/actors/hyku_addons/actors/json_fields_actor.rb

2. Do we really need to delete it if the version in hyrax-orcid is direct copy of the one in hyku_addons.
If we need to adjust that actor for hyku_addons do we need to always refer to hyrax-orcid

File: app/presenters/concerns/hyku_addons/person_or_organization_presenter_behavior.rb [hunk 2]
```diff
@@ -81,12 +83,13 @@ def wikidata_link
...
-        return [] unless send(field)&.first.present?
+        return if (field_value = solr_document.public_send(field).first.presence || "[]").blank?
 
-        JSON.parse(send(field).first).collect do |hash|
+        JSON.parse(field_value).collect do |hash|
           name = hash.slice("#{field}_family_name", "#{field}_given_name", "#{field}_organization_name").values.map(&:presence).compact.join(', ')
           PersonOrOrganization.new(name, hash["#{field}_orcid"], hash["#{field}_isni"], hash["#{field}_ror"], hash["#{field}_grid"], hash["#{field}_wikidata"])
         end
       end
+      alias participants person_or_organization_list
```
Comment (mankind) [inline app/presenters/concerns/hyku_addons/person_or_organization_presenter_behavior.rb:93]: **person_or_organization_list** defined elsewhere or it is being defined here for the first time. If so why rename it with alias, while not call it particpants

Comment (prdanelli) [reply app/presenters/concerns/hyku_addons/person_or_organization_presenter_behavior.rb:93]: the method `participants` is defined inside of Hyrax Orcid, but some essentialy the same thing as `person_or_organization_list`, so i just aliased it to the existing method here.

File: app/services/bolognese/writers/orcid/hyku_addons_xml_builder.rb [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+            xml_subtitle
+          end
+
+          xml_short_description
+          xml_work_type
+          xml_date_published
+
+          # NOTE: A full list of external-id-type: https://pub.orcid.org/v2.1/identifiers
+          @xml[:common].send("external-ids") do
```
Comment (mankind) [inline app/services/bolognese/writers/orcid/hyku_addons_xml_builder.rb:21]: Do you want to consider extracting this and the block underneath to standalone methods and call those here. 
It follows the pattern of the methods calls above where you called things like xml_date_published.

Additionally, it meant eliminate the need for rubocop override **rubocop:disable Metrics/MethodLength**

**Review by okeen — DISMISSED** [review]
This is great work, mate.

File: app/services/bolognese/writers/orcid/hyku_addons_xml_builder.rb (hunk 3 above)
Comment (prdanelli) [reply app/services/bolognese/writers/orcid/hyku_addons_xml_builder.rb:21]: As much as I hate changing structure to appease Rubocop, I agree that isn't bad enough for me not to do it :)

File: app/helpers/hyku_addons/orcid_helper_behavior.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  module OrcidHelperBehavior
+    extend ActiveSupport::Concern
+
+    include Hyrax::Orcid::HelperBehavior
+
+    # Override the Hyrax::Orcid JSON Fields creator/contributor helper, method which uses different participant name keys
+    def participant_to_string(type, participants)
+      return "-" if participants.blank?
+
+      participants
```
[the remark this answers is not in the document]
Comment (prdanelli) [reply app/helpers/hyku_addons/orcid_helper_behavior.rb:12]: I think the guard clause being here means i don't need to add a fallback within the lower block.

**Pull request merged**